pyuvdata/uvbeam.py

```diff
@@ -828,49 +829,56 @@ def _interp_az_za_rect_spline(self, az_array, za_array, freq_array):
         else:
             interp_basis_vector = None
 
+        # Npols is only defined for power beams.  For E-field beams need Nfeeds.
+        if self.beam_type == 'power':
+            Npol_feeds = self.Npols
+        else:
+            Npol_feeds = self.Nfeeds
+        splines_saved = hasattr(self, 'saved_interp_functions')
+
+        def get_lambda(real_lut, imag_lut=None):
+            if imag_lut is None:
+                return lambda za, az: real_lut(za, az, grid=False)
+            else:
+                return lambda za, az: (real_lut(za, az, grid=False) + 1j * imag_lut(za, az, grid=False))
```

Comment:
These spines need to be calculated and saved separately for the different vector axes (Naxes_vec) and spws (Nspws). I'm a little worried that it's passing the tests using the same spline for the different vector axes, that suggests inadequate testing.

Reply:
This function just returns the lambda functions, which get stored in an array with the full shape `(Naxes_vec, Nspws, Npol_feeds, Nfreqs)`. The resulting array does have separate splines saved along the different axes.
Right now the reuse test just checks for a power beam, but I'll add one for an E-field beam as well.

Reply:
ok, but I'm worried that they were pulled out of the for loops over those axes -- I think we needed those for loops to make sure the splines did not interact between those axes.

Reply:
It should just be this small function definition that's outside of the for loops. The splines are still made, stored, and used within those loops.

Reply:
Oh, I see. Sorry, I missed that.